app/com/lucidchart/open/cashy/controllers/BrowseController.scala

```diff
@@ -148,6 +156,66 @@ class BrowseController extends AppController {
   }
 
   /**
+   * Marks a hidden asset or folder as not hidden
+   */
+  def show(bucket: String, key: String) = AuthAction.authenticatedUser { implicit user =>
+    Action { implicit request =>
+      try {
+        if (key.endsWith("/")) {
+          // It is a folder
+          FolderModel.findByKey(bucket, key) match {
+            case Some(folder) => FolderModel.updateHidden(folder.id, false)
+            case None => FolderModel.createFolder(bucket, key, false)
```

Comment:
Why is it that the None case for folders create a folder, but the None case for assets throws an exception?